### PR TITLE
[Merged by Bors] - feat: port Data.Real.CauSeqCompletion

### PR DESCRIPTION
**Pull request by eric-wieser**
Unreverts #1440

---

- [x] depends on: leanprover-community/mathlib#18122
- [x] depends on: mathport output being available for the result after the above

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib#18122
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib#18122~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Real/CauSeqCompletion.lean [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+/-- Show the first 10 items of a representative of this equivalence class of cauchy sequences.
+
+The representative chosen is the one passed in the VM to `quot.mk`, so two cauchy sequences
+converging to the same number may be printed differently.
+-/
+unsafe instance [Repr β] : Repr (Cauchy abv) where
+  reprPrec r _ :=
+    let N := 10
+    let seq := r.unquot
+    "(sorry /- " ++ Std.Format.joinSep ((List.range N).map <| repr ∘ seq) ", " ++ ", ... -/)"
```
Comment (eric-wieser) [inline Mathlib/Data/Real/CauSeqCompletion.lean:291]: This was not present in the original PR

File: Mathlib/Data/Real/CauSeqCompletion.lean [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+section
+
+variable (β : Type _) [Ring β] (abv : β → α) [IsAbsoluteValue abv]
+
+/-- A class stating that a ring with an absolute value is complete, i.e. every Cauchy
+sequence has a limit. -/
+class IsComplete : Prop where
+  /-- Every Cauchy sequence has a limit. -/
+  is_complete : ∀ s : CauSeq β abv, ∃ b : β, s ≈ const abv b
+#align cau_seq.is_complete CauSeq.IsComplete
```
Comment (eric-wieser) [inline Mathlib/Data/Real/CauSeqCompletion.lean:322]: This is verbatim from the original PR, but is wrong; either the field should be called `IsComplete`, or the align statement should be `is_complete`.

Which is preferred?

Comment (jcommelin) [reply Mathlib/Data/Real/CauSeqCompletion.lean:322]: I think the field should be calles `isComplete`

Comment (jcommelin) [reply Mathlib/Data/Real/CauSeqCompletion.lean:322]: Do we need two `#align` statements? One for the class and one for the field?

**Review by Ruben-VandeVelde — APPROVED** [review]
I really don't know why we had to revert https://github.com/leanprover-community/mathlib4/pull/1440 over this, but it seems good to go now

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Data/Real/CauSeqCompletion.lean [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+  inv_zero := inv_zero
+  mul_inv_cancel x := CauSeq.Completion.mul_inv_cancel
+  ratCast q := ofRat q
+  ratCast_mk n d hd hnd := by rw [← ofRat_ratCast, Rat.cast_mk', ofRat_mul, ofRat_inv]; rfl
+
+theorem ofRat_div (x y : β) : ofRat (x / y) = (ofRat x / ofRat y : Cauchy abv) := by
+  simp only [div_eq_mul_inv, ofRat_inv, ofRat_mul]
+#align cau_seq.completion.of_rat_div CauSeq.Completion.ofRat_div
+
+-- Porting note: removed
```
Comment (eric-wieser) [inline Mathlib/Data/Real/CauSeqCompletion.lean:282]: @Ruben-VandeVelde: why did you remove this?

Comment (eric-wieser) [reply Mathlib/Data/Real/CauSeqCompletion.lean:282]: #1524

**Comment by eric-wieser** [comment]
> I really don't know why we had to revert #1440 over this, but it seems good to go now

Because:
* I am not setup to run mathport manually
* CI mathport runs against mathlib4 master, and I wanted the output from before #1440 was merged.

**Pull request closed without merging**